src/Symfony/Component/DependencyInjection/Compiler/ResolveDefinitionTemplatesPass.php

```diff
@@ -102,26 +102,6 @@ private function doResolveDefinition(ChildDefinition $definition)
         $def->setLazy($parentDef->isLazy());
         $def->setAutowired($parentDef->getAutowired());
 
-        self::mergeDefinition($def, $definition);
-
-        // merge autowiring types
-        foreach ($definition->getAutowiringTypes(false) as $autowiringType) {
-            $def->addAutowiringType($autowiringType);
-        }
-
-        // these attributes are always taken from the child
-        $def->setAbstract($definition->isAbstract());
-        $def->setShared($definition->isShared());
-        $def->setTags($definition->getTags());
-
-        return $def;
-    }
-
-    /**
-     * @internal
-     */
-    public static function mergeDefinition(Definition $def, ChildDefinition $definition)
-    {
```

Comment:
This class was simply changed to look like it did before #21530 (adjusted for changes since then)